demos/websocket_demo.cc

```diff
@@ -37,6 +37,18 @@ int main(int argc, char** argv) {
     app.run(argc, argv, [] () -> seastar::future<> {
         return async([] {
             websocket::server ws;
+            ws.register_handler("echo", [] (input_stream<char>& in,
+                        output_stream<char>& out) {
+                return do_with(false, [&in, &out] (bool done) {
```

Comment:
Um, how does this loop ever end? You never set `done` to `true`. Do you rely on the fact that `in.read()` propagates an exception when called on a closed stream? What can be done here instead is reading until EOF. Pseudocode below:

```cpp
return do_until([&in] {return in.eof();}, [&in, &out] {
    return in.read().then([&out] (temporary_buffer<char> f) {
        return out.write(std::move(f)).then([&out] {
            return out.flush();
        });
    });
});
```